src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+        if (radius < fabsf(planeDist) || VIA_FLAG_TEST(poly->flags_vIA, xpFlags)) {
+            if (SS_NULL == curNode->next) {
+                break;
+            } else {
+                curNode = &dyna->polyNodes.tbl[curNode->next];
+                continue;
+            }
+        }
+
+        acSizeReciprocal = 1.0f / ac_size;
+        temp_f18 = fabsf(nx) * acSizeReciprocal;
+        if (temp_f18 < 0.4f) {
+            if (SS_NULL == curNode->next) {
+                break;
+            } else {
+                curNode = &dyna->polyNodes.tbl[curNode->next];
+                continue;
+            }
+        }
+
+        // compute poly xMin/xMax
+        xMin = xMax = dyna->vtxList[VTX_INDEX(poly->flags_vIA)].x;
+        xTemp = dyna->vtxList[VTX_INDEX(poly->flags_vIB)].x;
+
+        if (xTemp < xMin) {
+            xMin = xTemp;
+        } else if (xMax < xTemp) {
+            xMax = xTemp;
+        }
+        xTemp = dyna->vtxList[poly->vIC].x;
+        if (xTemp < xMin) {
+            xMin = xTemp;
+        } else if (xMax < xTemp) {
+            xMax = xTemp;
+        }
+
+        xMin -= radius;
+        xMax += radius;
+        if (resultPos.x < xMin || xMax < resultPos.x) {
+            if (SS_NULL == curNode->next) {
+                break;
+            } else {
+                curNode = &dyna->polyNodes.tbl[curNode->next];
+                continue;
+            }
+        }
+        // 7a2c
+        if (func_80038F60(poly, dyna->vtxList, pos->y, resultPos.z, &intersect)) {
+            xIntersectDist = intersect - resultPos.x;
+            if (fabsf(xIntersectDist) <= radius / temp_f18) {
+                if (xIntersectDist * nx <= 4.0f) {
+                    // ab7ac8
+                    if (func_80039A3C(colCtx, poly, &resultPos.x, &resultPos.z, nx, ny, nz, acSizeReciprocal, planeDist,
+                                      radius, outPoly)) {
+                        *outBgId = bgId;
+                    }
+                    result = true;
+                }
+            }
+        }
+        if (SS_NULL == curNode->next) {
+            break;
+        }
+        curNode = &dyna->polyNodes.tbl[curNode->next];
+    }
+    *outX = resultPos.x;
+    *outZ = resultPos.z;
+    return result;
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80040284.s")
+#endif
+
+// Performs collision detection on all dyna poly walls using sphere `pos`, `radius`
+// returns true if a collision was detected
+// `outX` `outZ` return the displaced x,z coordinates
+// `outPoly` returns the pointer to the nearest poly collided with, or NULL
+// `outBgId` returns the index of the BgActor that owns `outPoly`
+// If `actor` is not NULL, an BgActor bound to that actor will be ignored
+s32 func_800409A8(CollisionContext* colCtx, u16 xpFlags, f32* outX, f32* outZ, Vec3f* pos, f32 radius,
+                  CollisionPoly** outPoly, s32* outBgId, Actor* actor);
+#ifdef NON_MATCHING
+// Codegen Issues
```

Comment:
can you be more specific in the comment, what type of issues

Reply:
not really, it's a big function and there's lots of regalloc issues

Reply:
regalloc is alot more descriptive than codegen issues